tests/testthat/test-http-libcurl.R

```diff
@@ -0,0 +1,26 @@
+test_that("basic HTTP methods work", {
+  withr::local_options(rsconnect.http = "libcurl")
+
+  test_http_GET()
+  test_http_POST_JSON()
+  test_http_POST_empty()
+  test_http_POST_file()
+  test_http_headers()
+})
+
+test_that("can stream PUT content from disk", {
```

Comment:
Future: Test PUT, PUT_JSON, DELETE across all transports.

Reply:
Future: Validate that the correct content-type header is set.

Reply:
I only tested `PUT` for libcurl because it's the only wrapper that has a custom path for PUT compared to the other methods.